src/SonarScanner.MSBuild.Shim/SonarEngineWrapper.cs

```diff
@@ -84,9 +85,10 @@ private bool Execute(ProcessRunnerArguments args)
         return result.Succeeded;
     }
 
-    private static IEnumerable<ProcessRunnerArguments.Argument> JavaParams(AnalysisConfig config)
+    private IEnumerable<ProcessRunnerArguments.Argument> JavaParams(AnalysisConfig config, IAnalysisPropertyProvider userCmdLineArguments)
```

Comment:
This whole method needs some comments in each section about what the section is about (You may want to add example properties for illustration).